### PR TITLE
FAT2-149 -  show message if last start date passed

### PR DESCRIPTION
**Pull request by robyoung91**

### CODE REVIEW

**Review by PeterDavidWilliams — DISMISSED** [review]

**Review by PeterDavidWilliams — APPROVED** [review]

File: src/SFA.DAS.FAT.Web.AcceptanceTests/Features/CourseDetail.feature [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿Feature: CourseDetail
+	As an employer
+	I want to know if the start date has passed for a bookmarked course
+	So that I can know it is no longer available
+
+Scenario: Navigate to course detail page for expired course
+When I navigate to the following url: /courses/101
+Then an http status code of 200 is returned
+And there is a message and button to course list displayed if course last start date has passed
```
Comment (dashton82) [inline src/SFA.DAS.FAT.Web.AcceptanceTests/Features/CourseDetail.feature:9]:
```suggestion
And there is a message and button to go to course list displayed
```

File: src/SFA.DAS.FAT.Web.AcceptanceTests/Steps/CourseDetailSteps.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    public sealed class CourseDetailSteps
+    {
+        private readonly ScenarioContext _context;
+
+        public CourseDetailSteps(ScenarioContext context)
+        {
+            _context = context;
+        }
+
+        [Then("there is a message and button to course list displayed if course last start date has passed")]
```
Comment (dashton82) [inline src/SFA.DAS.FAT.Web.AcceptanceTests/Steps/CourseDetailSteps.cs:25]: Will need to be changed to reflect the above comment

File: src/SFA.DAS.FAT.Web.AcceptanceTests/Steps/CourseDetailSteps.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        public async Task ThenThePageContentIncludesTheFollowing()
+        {
+            var response = _context.Get<HttpResponseMessage>(ContextKeys.HttpResponse);
+
+            var actualContent = await response.Content.ReadAsStringAsync();
+
+            var json = DataFileManager.GetFile("course-expired.json");
+            var expectedApiResponse = JsonConvert.DeserializeObject<TrainingCourse>(json);
+
+            if (DateTime.Now > expectedApiResponse.Course.StandardDates.LastDateStarts)
```
Comment (dashton82) [inline src/SFA.DAS.FAT.Web.AcceptanceTests/Steps/CourseDetailSteps.cs:35]: I'd remove this, we'd want it to fail if it doesn't show the below message

File: src/SFA.DAS.FAT.Web/Models/CourseViewModel.cs [hunk 4]
```diff
@@ -35,8 +38,10 @@ public class CourseViewModel
                 OverviewOfRole = course.OverviewOfRole,
                 TypicalJobTitles = string.IsNullOrEmpty(course.TypicalJobTitles) ? new List<string>() : course.TypicalJobTitles.Split("|").ToList(),
                 TypicalDuration = course.TypicalDuration,
-                MaximumFunding = course.MaxFunding.ToGdsCostFormat()
+                MaximumFunding = course.MaxFunding.ToGdsCostFormat(),
+                LastDateStarts = course.StandardDates?.LastDateStarts,
+                AfterLastStartDate = DateTime.Now > course.StandardDates?.LastDateStarts,
```
Comment (dashton82) [inline src/SFA.DAS.FAT.Web/Models/CourseViewModel.cs:43]: Use `DateTime.UtcNow`

**Review by dashton82 — CHANGES_REQUESTED** [review]

**Review by dashton82 — APPROVED** [review]

**Pull request merged**